src/main/scala/com/rbmhtechnology/eventuate/chaos/ChaosCounterInterface.scala

```diff
@@ -1,8 +1,31 @@
 package com.rbmhtechnology.eventuate.chaos
 
-import com.rbmhtechnology.eventuate.crdt.CounterService
+import com.rbmhtechnology.eventuate.crdt._
+
+
+object ChaosCounterInterface {
+
+  def apply(service: CounterService[Int]): ChaosCounterInterface = new ChaosCounterInterface(Left(service))
+
+  def apply(service: pure.CounterService[Int]): ChaosCounterInterface = new ChaosCounterInterface(Right(service))
+
+  implicit class EitherCounterService[A](service: Either[CounterService[A],pure.CounterService[A]]){
```

Comment:
Maybe use an adapter instead of `Either`. Something like:
```
object ChaosCounterInterface {

  def apply(service: CounterService[Int]): ChaosCounterInterface = new ChaosCounterInterface(plainCounterService(service))

  def apply(service: pure.CounterService[Int]): ChaosCounterInterface = new ChaosCounterInterface(pureCounterService(service))

  trait ChaosCounterService[A] {
    def update(id: String, delta: A): Future[A]

    def value(id: String): Future[A]
  }
  def plainCounterService[A](service: CounterService[A]): ChaosCounterService[A] = new ChaosCounterService[A] {
    override def update(id: String, delta: A): Future[A] = service.update(id, delta)
    override def value(id: String): Future[A] = service.value(id)
  }
  def pureCounterService[A](service: pure.CounterService[A]): ChaosCounterService[A] = new ChaosCounterService[A] {
    override def update(id: String, delta: A): Future[A] = service.update(id, delta)
    override def value(id: String): Future[A] = service.value(id)
  }
}
```
Avoids implicit conversions and could be extended more easily.